### PR TITLE
attempt to connect viewSwitcher onto redux store

### PR DESCRIPTION
**Pull request by mistvan**

### CODE REVIEW

File: src/actions/app.ts [hunk 1]
```diff
@@ -14,3 +14,15 @@ export function setDateRange(range: I.DateRange): I.Action {
...
+  return {
+    type: actionTypes.VIEW_TYPE_TILES
+  };
+}
+
+export function selectListDashboardView(): I.Action {
+  return {
+    type: actionTypes.VIEW_TYPE_LIST
+  };
+}
```
Comment (dullus) [inline src/actions/app.ts]: one action is enough when you alter same store property:
```ts
export function setDashboardView(dashboardViewType: ViewType): I.Action {
  return {
    payload: { dashboardViewType },
    type: actionTypes.VIEW_TYPE
  };
}
```

File: src/reducers/appReducer.ts [hunk 2]
```diff
@@ -30,6 +32,16 @@ export const app: Reducer<I.StateApp | undefined, I.Action> = (state = initialSt
...
+    case actionTypes.VIEW_TYPE_TILES:
+      return {
+        ...state,
+        dashboardViewType: ViewType.TILES
+      };
+    case actionTypes.VIEW_TYPE_LIST:
+      return {
+        ...state,
+        dashboardViewType: ViewType.LIST
+      };
```
Comment (dullus) [inline src/reducers/appReducer.ts]: One reducer should be enough:
```ts
case actionTypes.VIEW_TYPE:
  if (utils.isInterface<{ dashboardViewType: ViewType }>(action.payload, ['dashboardViewType'])) {
    return {
      ...state,
      dashboardViewType: action.payload.dashboardViewType
    };
  }
  return state;
```

File: src/components/viewSwitcher/ViewSwitcher.container.tsx [hunk 3]
```diff
@@ -0,0 +1,16 @@
...
+import ViewSwitcher, { IProps } from './ViewSwitcher';
+import { selectTilesDashboardView, selectListDashboardView } from '../../actions/app';
+
+const mapStateToProps = (state: I.State, ownProps: IProps): Partial<IProps> => ({
+  viewType: state.app.dashboardViewType
+});
+
+const mapDispatchToProps = (dispatch: Dispatch): IProps => ({
+  tilesViewSelect: () => dispatch(selectTilesDashboardView()),
+  listViewSelect: () => dispatch(selectListDashboardView())
```
Comment (dullus) [inline src/components/viewSwitcher/ViewSwitcher.container.tsx]: use single action:
```ts
tilesViewSelect: () => dispatch(setDashboardView(ViewType.TILES)),
listViewSelect: () => dispatch(setDashboardView(ViewType.LIST))
```

File: src/components/viewSwitcher/ViewSwitcher.stories.tsx [hunk 4]
```diff
@@ -1,11 +1,32 @@
...
-import ViewSwitcher, { ViewMode } from './ViewSwitcher';
+import ViewSwitcher, { ViewType } from './ViewSwitcher';
+import { action } from '@storybook/addon-actions';
+import { fakeStore } from '../fakeStore';
+import { Provider } from 'react-redux';
+import { initialState } from '../../reducers/appReducer';
 
-const viewSwitcherProps = {
-  mode: ViewMode.tiles
+const reducers = {
```
Comment (dullus) [inline src/components/viewSwitcher/ViewSwitcher.stories.tsx]: You don't need use container and fake storage in this story. Just pass parameters to component directly. You are not testing behavior but displaying only. Revert everything back as it was.

File: src/components/viewSwitcher/ViewSwitcher.tsx [hunk 5]
```diff
@@ -1,43 +1,56 @@
-import React from 'react';
+import React, { Component } from 'react';
 import { Button } from '../button/Button';
+import * as I from '../../lib/interfaces';
 import styles from './ViewSwitcher.module.css';
+import { VIEW_TYPE } from '../../lib/enums';
 
-interface IStats {
-  mode: ViewMode;
+export interface IProps extends Partial<I.StateApp> {
```
Comment (dullus) [inline src/components/viewSwitcher/ViewSwitcher.tsx]: this extension is overkill, I doubt you need everything from I.StateApp

File: src/reducers/appReducer.ts [hunk 6]
```diff
@@ -2,13 +2,15 @@ import { Reducer } from 'redux';
...
+import { VIEW_TYPE } from '../lib/enums';
 
 const { endDate, startDate } = utils.getDefaultDataRange();
 
 export const initialState: I.StateApp = {
   dateRangeFormVisible: false,
   endDate,
-  startDate
+  startDate,
+  dashboardViewType: VIEW_TYPE.tiles
```
Comment (dullus) [inline src/reducers/appReducer.ts]: sort alphabetically please

File: src/components/viewSwitcher/ViewSwitcher.tsx (hunk 5 above)
Comment (mistvan) [reply src/components/viewSwitcher/ViewSwitcher.tsx]: I tried to fix

File: src/components/viewSwitcher/ViewSwitcher.tsx [hunk 7]
```diff
@@ -1,43 +1,56 @@
-import React from 'react';
+import React, { Component } from 'react';
 import { Button } from '../button/Button';
+import * as I from '../../lib/interfaces';
 import styles from './ViewSwitcher.module.css';
+import { VIEW_TYPE } from '../../lib/enums';
 
-interface IStats {
-  mode: ViewMode;
+export interface IProps extends Partial<I.ViewSwitcher> {
```
Comment (dullus) [inline src/components/viewSwitcher/ViewSwitcher.tsx]: Still not good. 
1. Why you need extend I.ViewSwitcher interface when you define viewType in next line anyway?
2. I doubt that I.ViewSwitcher is needed somewhere else.

So remove I.ViewSwitcher, nobody needs that and also remove extension here, viewType is solely internal property of internal IProps interface. Interfaces.ts should be used for shared interfaces that many files are using, not for private internal interfaces of modules.

Comment (mistvan) [reply src/components/viewSwitcher/ViewSwitcher.tsx]: I think I understand, so I removed extension and created simple IProps object.

**Pull request merged**